### PR TITLE
Made X button unique in Persist Vehicle.

### PR DESCRIPTION
**Pull request by gir489returns**
Fixed issue reported by @lonelybud.

### CODE REVIEW

**Comment by lonelybud** [comment]
it is working fine. Files can be deleted this way. Folder need to be deleted manually.

File: src/views/vehicle/spawn/view_persist_car.cpp [hunk 1]
```diff
@@ -135,7 +135,7 @@ namespace big
 					}
 
 					ImGui::SameLine();
-					if (ImGui::SmallButton("X"))
+					if (ImGui::SmallButton(std::string("X##" + pair).c_str()))
 					{
 						file_name_to_delete = pair;
 					}
```
Comment (Yimura) [inline src/views/vehicle/spawn/view_persist_car.cpp]: Probably best to use ImGui::PushID and ImGui::PopID functions.

```cpp
ImGui::PushID(pair_lower.c_str());
ImGui::SameLine();
if (ImGui::SmallButton("X"))
{
	file_name_to_delete = pair;
}
ImGui::PopID();
```

And example code in the codebase:
https://github.com/YimMenu/YimMenu/blob/990946119f81420bc1d38d0f1a2d0b580dd6ad82/src/views/settings/view_reaction_settings.cpp#L9-L12

**Review by Yimura — CHANGES_REQUESTED** [review]

File: src/views/vehicle/spawn/view_persist_car.cpp (hunk 1 above)
Comment (gir489returns) [reply src/views/vehicle/spawn/view_persist_car.cpp:141]: There is no int to use that's readily available, I'd have to make one, for something that is only used situationally.

Comment (Yimura) [reply src/views/vehicle/spawn/view_persist_car.cpp]: PushID supports char array and int.

Comment (gir489returns) [reply src/views/vehicle/spawn/view_persist_car.cpp]: So it does. Fixed.

**Review by Yimura — APPROVED** [review]

**Pull request merged**